presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -515,6 +540,32 @@ private static LongWriteFunction timestampWithTimeZoneWriteFunction()
         };
     }
 
+    private static ObjectReadFunction longTimestampWithTimeZoneReadFunction()
+    {
+        return ObjectReadFunction.of(
+                LongTimestampWithTimeZone.class,
+                (resultSet, columnIndex) -> {
+                    // PostgreSQL does not store zone information in "timestamp with time zone" data type
+                    OffsetDateTime offsetDateTime = resultSet.getObject(columnIndex, OffsetDateTime.class);
+                    return LongTimestampWithTimeZone.fromEpochSecondsAndFraction(
+                            offsetDateTime.toEpochSecond(),
+                            (long) offsetDateTime.getNano() * PICOSECONDS_PER_NANOSECOND,
+                            UTC_KEY);
+                });
+    }
+
+    private static ObjectWriteFunction longTimestampWithTimeZoneWriteFunction()
+    {
+        return ObjectWriteFunction.of(
+                LongTimestampWithTimeZone.class,
+                (statement, index, value) -> {
+                    // PostgreSQL does not store zone information in "timestamp with time zone" data type
+                    long epochSeconds = value.getEpochMillis() / MILLISECONDS_PER_SECOND;
+                    long nanosOfSecond = value.getEpochMillis() % MICROSECONDS_PER_MILLISECOND * NANOSECONDS_PER_MILLISECOND + value.getPicosOfMilli() / PICOSECONDS_PER_NANOSECOND;
```

Comment:
MICROSECONDS_PER_MILLISECOND -> MILLISECONDS_PER_SECOND

@martint is the `%` and `/` right here? or floorDiv instead?

do we have helper methods somewhere so that we do not have to ask the above question ever again?